src/shogun/mathematics/linalg/internal/implementation/VectorSum.h

```diff
@@ -76,6 +76,40 @@ struct vector_sum
 	static T compute(Vector a);
 };
 
+/**
+ * @brief Specialization of generic vector_sum for the Native backend
+ */
+template<> template <class Vector>
+struct vector_sum<Backend::NATIVE, Vector>
+{
+	/** Scalar type */
+	typedef typename Vector::Scalar T;
+	
+	/**
+	 * Method that computes the sum of SGVectors
+	 * 
+	 * @param vec a vector whose sum has to be computed
+	 * @return the vector sum \f$\sum_i a_i\f$
+	 */
+	static T compute(SGVector<T> vec)
+	{
+		REQUIRE(vec.vlen > 0, "Vector can not be empty!\n");
+		return compute(vec.vector,vec.vlen);
+	}
+
+	/**
+	 * Method that computes the sum of SGVectors
+	 * 
+	 * @param vec a vector whose sum has to be computed
+	 * @param len the length of the vector
+	 * @return the vector sum \f$\sum_i a_i\f$
+	 */
+	static T compute(T* vec, index_t len)
+	{
+		return std::accumulate(vec,vec+len,0);
```

Comment:
Guys no worries I'll commit this directly, `#include <numeric>` solves the problem